### PR TITLE
(SUP-2678) Update Readme ca_extend::check_agent_expiry task runs against primary server and not agent nodes

### LINKED ISSUE #52: ca_extend::check_agent_expiry task runs against primary server and not agent nodes
## Describe the Change You Would Like
Currently, the readme says:
```
* ca_extend::check_agent_expiry
Checks if any agent certificate expires by a certain date. Defaults to three months from today.
```
and provides the following example for usage
```
bolt task run ca_extend::check_agent_expiry --targets <TargetSpec>
```
as per code for this task, the only valid `<TargetSpec>` is the `primary` server either by using the `localhost` ref or its `fqdn`, if run remotely. This needs to be clarified in more detailed so one doesn't replace `<TargetSpec>` by agent nodes which is a valid ref for the 2 previous examples given in the documentation:

```
bolt plan run ca_extend::upload_ca_cert cert=<path_to_cert> --targets <TargetSpec>
bolt task run ca_extend::check_ca_expiry --targets <TargetSpec>
```

### PR DESCRIPTION
**Pull request by jordi-garcia**
[ca_extend::check_agent_expiry task runs against primary server and not agent nodes](https://github.com/puppetlabs/ca_extend/issues/52)
Update to readme file for clarity and removal of $PUPPET_BIN/ as redundant and prone to mislead

### CODE REVIEW

**Review by MartyEwings — COMMENTED** [review]
@jordi-garcia typically we put the fully qualified path to the Puppet Binary in the scripts for OS compatibility, is there an issue you are seeing specifically with declaring it?

File: README.md [hunk 1]
```diff
@@ -177,16 +178,49 @@ Note that if you are running `extend_ca_cert` locally on the primary Puppet serv
...
 bolt task run ca_extend::check_ca_expiry --targets <TargetSpec>
 ```
 
+### ca_extend::check_agent_expiry Task
+
+You can use this task to categorise all PE certs in a PE environment as part of a valid or expiring section based on a customasable date in the future (default 3 months from now). This task runs against a `primary` server and checks all cer `/etc/puppetlabs/puppet/ssl/ca/signed` as the single source of truth for the PE environment and splits the certs between a valid section or expiring section. 
+
 ```bash
-bolt task run ca_extend::check_agent_expiry --targets <TargetSpec>
+bolt task run ca_extend::check_agent_expiry --targets local://$(hostname -f)
```
Comment (MartyEwings) [inline README.md]: If the target can only be localhost, would it make sense to simplify this to "--targets localhost"

Comment (jordi-garcia) [reply README.md]: The readme mentions in the Usage section:
"Note that if you are running extend_ca_cert locally on the primary Puppet server, you can avoid potential Bolt transport issues by specifying --targets local://$(hostname -f)"

So, I went with that reference. Happy to update, if necessary

Comment (MartyEwings) [reply README.md]: @jordi if the docs say it lets go with that! id just not seen it before

Comment (MartyEwings) [reply README.md]: So per discussion local:// forces the local transport incase hosts are screwed up, but (hostname - f) introduces variability, i think we should just go local://localhost

Comment (m0dular) [reply README.md]: The `local://` part is a specific protocol that avoids trying to `ssh` into the box we're currently on, which has caused issues before.  I'm not sure if `$(hostname -f)` vs `localhost` would make much of a difference, but I think I'd prefer the `hostname` approach.  Someone could have a weird IPv6 `localhost` configuration or something, and the `-f` flag should be portable.  Except Solaris, let's not speak of that...

Comment (m0dular) [reply README.md]: Asked about this in #bolt, and it doesn't actually matter what the rest of the URI has, as long as the protocol is `local://`.

File: README.md [hunk 2]
```diff
@@ -166,27 +168,59 @@ First, check the expiration of the Puppet agent certificate by running the follo
...
+
+You can use this task to check the ca cert expiry on the `primary` mainly but you can also use it to check that a remote CA cert has been updated after using any means to update an expired CA cert on an agent.
+
 ```bash
 bolt task run ca_extend::check_ca_expiry --targets <TargetSpec>
 ```
 
+### ca_extend::check_agent_expiry Task
+
+You can use this task to categorise all PE certs in a PE environment as part of a valid or expiring section based on a customasable date in the future (default 3 months from now). This task runs against a `primary` server and checks all cer `/etc/puppetlabs/puppet/ssl/ca/signed` as the single source of truth for the PE environment and splits the certs between a valid section or expiring section.
```
Comment (MartyEwings) [inline README.md]: few spelling mistakes here to correct:

```
You can use this task to categorize all PE certs in a PE environment as part of a valid or expiring section based on a customizable date in the future (default 3 months from now). This task runs against a `primary` server and checks all cert `/etc/puppetlabs/puppet/ssl/ca/signed` as the single source of truth for the PE environment and splits the certs between a valid section or expiring section. 
```

**Review by MartyEwings — CHANGES_REQUESTED** [review]

**Pull request merged**